spectral_cube/spectral_cube.py

```diff
@@ -2823,8 +2830,8 @@ def apply_function_parallel_spectral(self,
             else:
                 pbu = object
 
-            for arg in spectra:
-                _apply_function(arg, outcube, function, **kwargs)
+            for arg in iteration_data:
+                applicator(arg, outcube, function, **kwargs)
                 pbu()
 
 
```

Comment:
`LazyComparisonMask` on `outcube` here?

Reply:
What do you have in mind?  Should we just be using a `LazyMask` that checks for finite values in the output cube?  Right now we're leaving it untouched, which isn't crazy.